File: app/src/main/java/com/example/retrofitmvvmsample/viewModel/UserViewModel.java
package com.example.retrofitmvvmsample.viewModel;

import android.app.Application;

import androidx.annotation.NonNull;
import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;

import com.example.retrofitmvvmsample.modelClass.Datum;
import com.example.retrofitmvvmsample.repo.UserLocalRepo;
import com.example.retrofitmvvmsample.repo.UsersNetworkRepo;

import java.util.List;

public class UserViewModel extends AndroidViewModel {
    private final UsersNetworkRepo usersNetworkRepo;
    private final UserLocalRepo userLocalRepo;

    public UserViewModel(@NonNull Application application) {
        super(application);
        usersNetworkRepo = UsersNetworkRepo.getInstance(application);
        userLocalRepo = UserLocalRepo.getInstance(application);
    }

    public LiveData<List<Datum>> getUsersResponse(int page) {
        usersNetworkRepo.getMatches(page);
        return userLocalRepo.getUsers();
    }
}

File: app/src/main/java/com/example/retrofitmvvmsample/viewModel/UserViewModel.kt
package com.example.retrofitmvvmsample.viewModel

import android.app.Application
import androidx.lifecycle.AndroidViewModel
import androidx.lifecycle.viewModelScope
import com.example.retrofitmvvmsample.repo.UsersRepo
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.withContext

class UserViewModel(application: Application, private val usersRepo: UsersRepo) :
    AndroidViewModel(application) {


    suspend fun getUsersResponse(page: Int) =
        withContext(viewModelScope.coroutineContext + Dispatchers.IO) {
            usersRepo.getUsers(page)
        }

}